onnxruntime/core/providers/cpu/nn/string_normalizer.cc

```diff
@@ -201,7 +201,16 @@ class Utf8Converter {
 
 #endif
 
-const std::string default_locale("en_US.UTF-8");  // All non-MS
+#if defined(__APPLE__)
```

Comment:
Would it be simpler to add a compile definition of `__IOS__` at the cmake level and use that in the `#ifdef` given cmake has an `IOS` variable that could be checked?  https://cmake.org/cmake/help/latest/variable/IOS.html

Reply:
if we go with this approach, should see if other uses of `<TargetConditionals.h>` can be replaced too. the name `__IOS__` is reserved as it has double underscores. we should probably pick something ORT specific like `ONNXRUNTIME_IOS`.

Reply:
thanks for the suggestions. 

looks like we do have a few places that uses <TargetConditionals.h> need to validate. For this pr's scope, will go with the current approach.